jwst/conftest.py

```diff
@@ -35,8 +35,13 @@ def mk_tmp_dirs():
 
 
 @pytest.fixture(autouse=True)
-def monkey_patch_s3_client(monkeypatch):
-    monkeypatch.setattr(s3_utils, "_CLIENT", lib_helpers.MockS3Client())
+def monkey_patch_s3_client(monkeypatch, request):
+    # If tmpdir is used in the test, then it is providing the file.  Map to it.
+    if "tmpdir" in request.fixturenames:
+        path = request.getfixturevalue("tmpdir")
```

Comment:
Should we create a special-purpose temp directory and fixture for this?  Something like `s3_root_dir`.  That would make what's going on clearer in the tests -- as it is, the implicit mapping of tmpdir to S3 URIs is a little mysterious.  It also may not always be desirable to treat all files in tmpdir like they're available from S3.